### PR TITLE
[FLINK-12704][python] Enable the configuration of using blink planner.

### PR DESCRIPTION
**Pull request by WeiZhong94**
## What is the purpose of the change

*This pull request adds the `EnvironmentSettings` class for flink python API to enable the configuration of using blink planner.*

## Brief change log

  - *Add `EnvironmentSettings` and relevant tests for flink python API.*
  - *Modify the create method of `StreamTableEnvironment` and `BatchTableEnvironment` to accept the `EnvironmentSettings` parameter.*
  - *Adjusted other interfaces of BatchTableEnvironment to match the behavior when using blink planner.`*

## Verifying this change

This change is already covered by existing tests, such as *test_environment_settings.py*, *test_environment_settings_completeness.py*, *test_table_environment_api.py*.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (python docs)

### CODE REVIEW

File: flink-python/pyflink/table/environment_settings.py [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+        ...     .use_old_planner() \\
+        ...     .in_streaming_mode() \\
+        ...     .with_built_in_catalog_name("my_catalog") \\
+        ...     .with_built_in_database_name("my_database") \\
+        ...     .build()
+    """
+
+    class Builder(object):
+        """
+        A builder for :class:`EnvironmentSettingss`.
```
Comment (dianfu) [inline flink-python/pyflink/table/environment_settings.py]: typo: EnvironmentSettingss -> EnvironmentSettings. EnvironmentSettingss also appear in other places, should be updated as well.

File: flink-python/pyflink/table/environment_settings.py [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+        def build(self):
+            """
+            Returns an immutable instance of EnvironmentSettings.
+
+            :return: an immutable instance of EnvironmentSettings.
+            :rtype: EnvironmentSettings
+            """
+            return EnvironmentSettings(self._j_builder.build())
+
+    def __init__(self, j_envrionment_settings):
```
Comment (dianfu) [inline flink-python/pyflink/table/environment_settings.py]: typo: j_envrionment_settings -> j_environment_settings

**Comment by WeiZhong94** [comment]
@dianfu Thanks for your review! I have fixed these spelling errors in the latest commit.

**Comment by dianfu** [comment]
Thanks a lot for the update. +1 from my side.

**Comment by WeiZhong94** [comment]
@sunjincheng121  Thanks for your review! I have added the end to end tests for the blink planner table environments(stream and batch) in the latest commit.

**Pull request closed without merging**